cmd/crank/beta/xpkg/init.go

```diff
@@ -116,14 +138,127 @@ func (c *initCmd) Run(k *kong.Context, logger logging.Logger) error {
 		return errors.Wrapf(err, "failed to get repository's HEAD from %q", repoURL)
 	}
 
-	// TODO(phisco): replace placeholders for the name all around the
-	// 	repository? Maybe we can just agree on some markdown text in the
-	// 	repos to print to let the user know what to do next?
-
 	_, err = fmt.Fprintf(k.Stdout, "Initialized package %q in directory %q from %s (%s)\n", c.Name, c.Directory, getPrettyURL(logger, repoURL, ref), ref.Name().Short())
+	if err != nil {
+		return errors.Wrap(err, "failed to write to stdout")
+	}
+
+	err = handleNotes(k.Stdout, logger, c.Directory)
+	if err != nil {
+		return errors.Wrap(err, "failed to handle NOTES.txt")
+	}
+
+	err = c.handleInitScript(k, logger, c.Directory, c.Name)
+	if err != nil {
+		return errors.Wrap(err, "failed to handle initialization script")
+	}
+
 	return err
 }
 
+// handleNotes prints the NOTES.txt file in the template
+// repository, if it exists.
+func handleNotes(w io.Writer, logger logging.Logger, dir string) error {
+	notesFile := dir + "/" + notes
+	f, err := os.Stat(notesFile)
+	switch {
+	case os.IsNotExist(err):
+		// no NOTES.txt file, skip
+		logger.Debug("No NOTES.txt found, skipping")
+		return nil
+	case err != nil:
+		return errors.Wrapf(err, "failed to stat notes file %s", notesFile)
+	case f.IsDir():
+		return errors.Errorf("%s is not a file", notesFile)
+	}
+
+	return errors.Wrapf(printFile(w, notesFile), "failed to print file %s", notesFile)
+}
+
+// handleInitScript runs the init.sh script in the template repository, if it
+// exists.
+func (c *initCmd) handleInitScript(k *kong.Context, logger logging.Logger, dir, name string) error {
```

Comment:
nit: looks like this function has a receiver only for access to the `RunInitScript` value. The caller is already passing in other members of the `initCmd` type, so it's mixing a bit here. Should we consider being consistent instead? (either don't pass in any members or pass them all in)

Reply:
Right, missed during the cleaning. Changed it to not pass any unneeded ones plus made handleNotes the same for consistency